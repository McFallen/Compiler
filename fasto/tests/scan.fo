fun int incr(int a, int b) = a + b

fun int writeInt(int n) = write(n)

fun [int] main() =
  let a = {5, 9, 11, 2} in
  let b = scan(incr, 1, a) in
  map(writeInt, b)
