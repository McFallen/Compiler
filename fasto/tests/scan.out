1 6 15 26 28 
